pcsx2/DEV9/InternalServers/DNS_Logger.cpp

```diff
@@ -0,0 +1,203 @@
+				opName = "Status";
+				break;
+			case DNS_OPCode::Reserved:
+				opName = "Reserved";
+				break;
+			case DNS_OPCode::Notify:
+				opName = "Notify";
+				break;
+			case DNS_OPCode::Update:
+				opName = "Update";
+				break;
+			default:
+				opName = "Unknown";
+				break;
+		}
+		Console.WriteLn("DEV9: DNS: OpCode %s (%i)", opName, dns->GetOpCode());
+		Console.WriteLn("DEV9: DNS: Is Authoritative (not cached)? %s", dns->GetAA() ? "True" : "False");
+		Console.WriteLn("DEV9: DNS: Is Truncated? %s", dns->GetTC() ? "True" : "False");
+		Console.WriteLn("DEV9: DNS: Recursion Desired? %s", dns->GetRD() ? "True" : "False");
+		Console.WriteLn("DEV9: DNS: Recursion Available? %s", dns->GetRA() ? "True" : "False");
+		Console.WriteLn("DEV9: DNS: Zero %i", dns->GetZ0());
+		Console.WriteLn("DEV9: DNS: Authenticated Data? %s", dns->GetAD() ? "True" : "False");
+		Console.WriteLn("DEV9: DNS: Checking Disabled? %s", dns->GetCD() ? "True" : "False");
+		const char* rName;
+		switch ((DNS_RCode)dns->GetRCode())
+		{
+			case DNS_RCode::NoError:
+				rName = "NoError";
+				break;
+			case DNS_RCode::FormatError:
+				rName = "FormatError";
+				break;
+			case DNS_RCode::ServerFailure:
+				rName = "ServerFailure";
+				break;
+			case DNS_RCode::NameError:
+				rName = "NameError";
+				break;
+			case DNS_RCode::NotImplemented:
+				rName = "NotImplemented";
+				break;
+			case DNS_RCode::Refused:
+				rName = "Refused";
+				break;
+			case DNS_RCode::YXDomain:
+				rName = "YXDomain";
+				break;
+			case DNS_RCode::YXRRSet:
+				rName = "YXRRSet";
+				break;
+			case DNS_RCode::NXRRSet:
+				rName = "NXRRSet";
+				break;
+			case DNS_RCode::NotAuth:
+				rName = "NotAuth";
+				break;
+			case DNS_RCode::NotZone:
+				rName = "NotZone";
+				break;
+			default:
+				rName = "Unknown";
+				break;
+		}
+		Console.WriteLn("DEV9: DNS: Result %s (%i)", rName, dns->GetRCode());
+
+		Console.WriteLn("DEV9: DNS: Question Count %i", dns->questions.size());
+		Console.WriteLn("DEV9: DNS: Answer Count %i", dns->answers.size());
+		Console.WriteLn("DEV9: DNS: Authority Count %i", dns->authorities.size());
+		Console.WriteLn("DEV9: DNS: Additional Count %i", dns->additional.size());
+
+		for (size_t i = 0; i < dns->questions.size(); i++)
+		{
+			DNS_QuestionEntry entry = dns->questions[i];
+			Console.WriteLn("DEV9: DNS: Q%i Name %s", i, entry.name.c_str());
+			Console.WriteLn("DEV9: DNS: Q%i Type %i", i, entry.entryType);
+			Console.WriteLn("DEV9: DNS: Q%i Class %i", i, entry.entryClass);
+		}
+		for (size_t i = 0; i < dns->answers.size(); i++)
+		{
+			DNS_ResponseEntry entry = dns->answers[i];
+			Console.WriteLn("DEV9: DNS: Ans%i Name %s", i, entry.name.c_str());
+			Console.WriteLn("DEV9: DNS: Ans%i Type %i", i, entry.entryType);
+			Console.WriteLn("DEV9: DNS: Ans%i Class %i", i, entry.entryClass);
+			Console.WriteLn("DEV9: DNS: Ans%i TTL %i", i, entry.timeToLive);
+
+			char* strArr = new char[entry.data.size() * 4 + 1];
```

Comment:
Perhaps a helper function that does something like this to keep code duplication down and make sure memory is properly deallocated?
```
std::string str;
//str.reserve(entry.data.size() * 4 + 1);  -- If this is a hot path, otherwise let std::string deal with the allocation
for (const auto& d : entry.data) {
  str += std::to_string(d) + ":";
}
return str;
```